microsoft/atlas-design
# Form behavior updates

Task: task-692607, task-687169, task-691911

Link: preview-435

This PR ensures that there is bottom margin for the form behavior error container for all scenarios. Bottom spacing was accidentally removed recently, which affected the no edit scenario. Logic is also added to dispatch the `form-validating` event for the no edit scenario. `.form-error-container` is replaced with atomics and data attribute `data-form-error-container`. Updated form behavior logic to use `data-field-error` to make `.field-error` solely for styling. Added a small section on custom validation in the documentation.

## Testing

1. Visit https://design.docs.microsoft.com/pulls/435/components/form.html . Scroll down to form behavior section. Inspect the form to open dev tools, remove the `new` attribute from the form behavior and submit the form. There should be some spacing below the error summary at the top of the form. Type in an input and submit. The same bottom spacing should still exist.
2. There should be a warning icon in the error summary container. 
![image](https://user-images.githubusercontent.com/94572161/184765405-e2996b4c-0513-4606-b070-f082350a41fb.png)
3. Make some inputs and submit again, should still work the same as before.
